src/foam/flow/PromiseSlot.js

```diff
@@ -0,0 +1,33 @@
+foam.CLASS({
+  package: 'foam.flow',
+  name: 'PromiseSlot',
+  implements: [ 'foam.core.Slot' ],
+  documentation: `
+    A slot that takes a promise and sets its value to its value when it
+    resolves.
+  `,
+  properties: [
+    {
+      name: 'promise',
+      postSet: function(_, n) {
+        n.then(function(v) {
```

Comment:
We could consider detaching everyone listening to us at this point, as a Promise can only resolve once and it could help prevent memory leaks if others forget to unsub, but maybe that's not necessary.

Reply:
We'll need to decide whether we want to support the ability to change the promise or not. If we don't care about supporting that we can detach listeners but if we want that, we can't. Currently, it supports the promise being able to change.